pkg/sql/sem/builtins/builtins.go

```diff
@@ -321,6 +321,53 @@ var builtins = map[string]builtinDefinition{
 				"Supports encodings 'UTF8' and 'LATIN1'.",
 		}),
 
+	// https://www.postgresql.org/docs/9.0/functions-binarystring.html#FUNCTIONS-BINARYSTRING-OTHER
+	"get_bit": makeBuiltin(tree.FunctionProperties{Category: categoryString},
+		tree.Overload{
+			Types:      tree.ArgTypes{{"val", types.VarBit}, {"position", types.Int}},
+			ReturnType: tree.FixedReturnType(types.Int),
+			Fn: func(_ *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				val := tree.MustBeDBitArray(args[0])
+				position := tree.MustBeDInt(args[1])
+
+				// Checking whether index asked is inside array of bits.
+				if position < 0 || uint(position) >= val.BitLen() {
+					return nil, pgerror.Newf(
+						pgcode.ArraySubscript, "bit index %d out of valid range (0..%d)", position, val.BitLen()-1)
+				}
+				return tree.NewDInt(tree.DInt(val.BitArray.String()[position] - '0')), nil
```

Comment:
[nit], but I find would find it clearer to case on the value in the bit array rather than doing a string cast into character math